pkg/agent/client.go

```diff
@@ -249,11 +252,35 @@ func (a *Client) Connect() (int, error) {
 	}
 	a.conn = conn
 	a.stream = stream
+	a.serverChannel = make(chan *client.Packet, xfrChannelSize)
+	a.serverError = nil
```

Comment:
serverError could deserve some comments about: why it doesn't need lock protection, and when set the client is effectively dead.

Reply:
Done.